### PR TITLE
checkout: Remove selectively on directory files when changed

### PR DESCRIPTION
**Pull request by mroutis**
Fix: #1220

### CODE REVIEW

File: tests/test_checkout.py [hunk 1]
```diff
@@ -143,6 +160,23 @@ def test(self):
         self.assertFalse(os.path.exists(fname))
 
 
+class TestCheckoutSelectiveRemove(CheckoutBase):
+    def test(self):
+        # Use copy to test for changes in the inodes
+        main(['config', 'cache.type', 'copy'])
```
Comment (efiop) [inline tests/test_checkout.py]:
```suggestion
        ret = main(['config', 'cache.type', 'copy'])
        self.assertEqual(ret, 0)
```

File: tests/test_checkout.py [hunk 2]
```diff
@@ -143,6 +160,23 @@ def test(self):
...
+
+        stage = self.dvc.add('data_dir')[0]
+        staged_files = self.outs_info(stage)
+
+        os.remove(staged_files[0].path)
+        main(['checkout', 'data_dir.dvc'])
```
Comment (efiop) [inline tests/test_checkout.py]:
```suggestion
        ret = main(['checkout', stage.relpath])
        self.assertEqual(ret, 0)
```

File: tests/test_checkout.py [hunk 3]
```diff
@@ -143,6 +160,23 @@ def test(self):
         self.assertFalse(os.path.exists(fname))
 
 
+class TestCheckoutSelectiveRemove(CheckoutBase):
+    def test(self):
+        # Use copy to test for changes in the inodes
+        main(['config', 'cache.type', 'copy'])
+
+        stage = self.dvc.add('data_dir')[0]
```
Comment (efiop) [inline tests/test_checkout.py]:
```suggestion
        stages = self.dvc.add(self.DATA_DIR)
        self.assertEqual(len(stages), 1)
        stage = stages[0]
```

File: tests/test_checkout.py [hunk 4]
```diff
@@ -143,6 +160,23 @@ def test(self):
...
+        staged_files = self.outs_info(stage)
+
+        os.remove(staged_files[0].path)
+        main(['checkout', 'data_dir.dvc'])
+
+        checkedout_files = self.outs_info(stage)
+
+        self.assertNotEqual(staged_files[0].inode, checkedout_files[0].inode)
```
Comment (efiop) [inline tests/test_checkout.py]:
```suggestion
        self.assertEqual(len(staged_files), len(checkedout_files))
        self.assertEqual(staged_files[0].path, checkedout_files[0].path)
        self.assertNotEqual(staged_files[0].inode, checkedout_files[0].inode)
```

File: tests/test_checkout.py [hunk 5]
```diff
@@ -4,9 +4,12 @@
 import stat
 import shutil
 import filecmp
+import itertools
```
Comment (efiop) [inline tests/test_checkout.py]: Not used.
```suggestion
```

File: tests/test_checkout.py (hunk 1 above)
Comment (mroutis) [reply tests/test_checkout.py]: @efiop, we can skip this assertion, as we are already testing the `config` command in `tests/test_config.py`.

File: tests/test_checkout.py (hunk 2 above)
Comment (mroutis) [reply tests/test_checkout.py]: This one too, there's already a test for the `checkout` command executed from the CLI

File: tests/test_checkout.py (hunk 1 above)
Comment (efiop) [reply tests/test_checkout.py]: We do, but it doesn't save from making a mistake here. Better safe than sorry :slightly_smiling_face:

**Comment by efiop** [comment]
@mroutis Let's add those asserts just in case.

File: tests/test_checkout.py (hunk 2 above)
Comment (efiop) [reply tests/test_checkout.py]: Same here. Could we add this as well please?

**Comment by efiop** [comment]
There is one more assert left in the comment above.

**Comment by mroutis** [comment]
Sorry, @efiop , missed that one, thanks for pointing that out :+1:

**Pull request merged**